transport/src/main/java/io/scalecube/transport/TransportImpl.java

```diff
@@ -169,50 +171,70 @@ public NetworkEmulator networkEmulator() {
         });
   }
 
-  private Mono<Channel> getOrConnect(Address address) {
+  private Mono<Connection> getOrConnect(Address address) {
+
     return Mono.create(
         sink ->
-            outgoingChannels
+            outgoingConnections
                 .computeIfAbsent(address, this::connect0)
                 .subscribe(sink::success, sink::error));
   }
 
-  private Mono<Channel> connect0(Address address) {
-    return connect1(address)
-        .doOnSuccess(
-            channel -> {
-              LOGGER.debug(
-                  "Connected from {} to {}: {}", TransportImpl.this.address, address, channel);
-
-              ChannelFutureListener closeFutureListener =
-                  future -> {
-                    LOGGER.debug("Disconnected from: {} {}", address, future.channel());
-                    outgoingChannels.remove(address);
-                  };
-
-              channel.closeFuture().addListener(closeFutureListener);
+  private Mono<? extends Connection> connect0(Address address) {
+    return TcpClient.create(ConnectionProvider.fixed("client-" + address, 1))
```

Comment:
`ConnectionProvider` is an abstraction of _connection pooling_ (previously it was `PoolResources` or something like that) .  TL;DR  we don't need it  -- let's preserve old approach and don't use any connection pooling et al. `TcpClient` has another one factory method -- `TcpClient.newConnection()`, use it.

Reply:
and all chanels would shut down when I call shutdown on LoopResources

Reply:
> and all chanels would shut down when I call shutdown on LoopResources

Try that. I did. On prev version on class BootstrapFactory I was calling shutdown. It wasn't acceptable because test passing time just doubled. But let's try with LoopResources.dispose().

Reply:
damn it, can't figure out the correct way... tests for ordered requests fail, all the other tests pass.